pkg/controller/plan/adapter/vsphere/builder.go

```diff
@@ -675,6 +686,9 @@ func (r *Builder) mapDisks(vm *model.VM, persistentVolumeClaims []*core.Persiste
 				},
 			},
 		}
+		if bootDisk == i+1 {
```

Comment:
comment? why this is a special case?

Reply:
in this case the root disk was changed we need to change the boot order, so this finds the new root disk location and sets its as the VM bootable disk.

Reply:
when writing how it's done it makes sense.
it still will be great to have a comment such as:
`if the selected boot device is the current disk, set it as the first in the boot order`